backend/src/adapters/rpc_funcs/utils_v2.py

```diff
@@ -0,0 +1,537 @@
+        df['tx_fee'] = (df['gas_price'] * df['gas_used']) / 1e18
+
+    if 'l1_fee' in df.columns and 'tx_fee' in df.columns:
+        df['tx_fee'] += df['l1_fee']
+    
+    return df
+
+def calculate_priority_fee(df):
+    if 'max_fee_per_gas' in df.columns and 'base_fee_per_gas' in df.columns:
+        df['priority_fee_per_gas'] = (df['max_fee_per_gas'] - df['base_fee_per_gas']) / 1e18
+        df.drop('base_fee_per_gas', axis=1, inplace=True)
+    else:
+        df['priority_fee_per_gas'] = np.nan
+    return df
+
+def handle_max_fee_per_blob_gas(df):
+    if 'max_fee_per_blob_gas' in df.columns:
+        df['max_fee_per_blob_gas'] = df['max_fee_per_blob_gas'].apply(
+            lambda x: int(x, 16) / 1e18 if isinstance(x, str) and x.startswith('0x') else float(x) / 1e18
+        ).astype(float)
+    return df
+
+# ---------------- Connection Functions ------------------
+def connect_to_node(rpc_config):
+    try:
+        return Web3CC(rpc_config)
+    except ConnectionError as e:
+        print(f"ERROR: failed to connect to the node with config {rpc_config}: {e}")
+        raise
+
+def connect_to_s3():
+    try:
+        aws_access_key_id = os.getenv("AWS_ACCESS_KEY_ID")
+        aws_secret_access_key = os.getenv("AWS_SECRET_ACCESS_KEY")
+        bucket_name = os.getenv("S3_LONG_TERM_BUCKET")
+
+        if not aws_access_key_id or not aws_secret_access_key or not bucket_name:
+            raise EnvironmentError("AWS access key ID, secret access key, or bucket name not found in environment variables.")
+
+        s3 = boto3.client('s3',
+                            aws_access_key_id=aws_access_key_id,
+                            aws_secret_access_key=aws_secret_access_key)
+        return s3, bucket_name
+    except Exception as e:
+        print("ERROR: An error occurred while connecting to S3:", str(e))
+        raise ConnectionError(f"An error occurred while connecting to S3: {str(e)}")
+
+def check_s3_connection(s3_connection):
+    return s3_connection is not None
+
+def s3_file_exists(s3, file_key, bucket_name):
+    try:
+        s3.head_object(Bucket=bucket_name, Key=file_key)
+        return True
+    except botocore.exceptions.ClientError as e:
+        error_code = int(e.response['Error']['Code'])
+        if error_code == 404:
+            return False
+        else:
+            raise e
+
+# ---------------- Generic Preparation Function ------------------
+def prep_dataframe_new(df, chain):
+    # Check for OP chains and adjust configuration accordingly
+    op_chains = ['zora', 'base', 'optimism', 'gitcoin_pgn', 'mantle', 'mode', 'blast', 'redstone', 'orderly', 'derive', 'karak', 'ancient8', 'kroma', 'fraxtal', 'cyber']
+    
+    if chain.lower() in op_chains:
+        config = chain_configs.get('op_chains')  # Use the specific 'op_chains' configuration
+    else:
+        config = chain_configs.get(chain.lower(), chain_configs['default'])  # Use the normal chain config
```

Comment:
let's not use any "default" - every chain should be assigned to a preparation step explicitly. If it isn't mentioned anywhere we raise an error